### PR TITLE
[CN-312] Detect member crash and trigger Force Start action

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

File: controllers/hazelcast/hot_backup_controller.go [hunk 1]
```diff
@@ -22,14 +23,16 @@ import (
 type HotBackupReconciler struct {
 	client.Client
 	Log       logr.Logger
+	Scheme    *runtime.Scheme
```
Comment (hakanmemisoglu) [inline controllers/hazelcast/hot_backup_controller.go:26]: I think that this field is not used.

Comment (hakanmemisoglu) [inline controllers/hazelcast/hot_backup_controller.go:26]: I think that this field is not used.

**Review by hakanmemisoglu — APPROVED** [review]
LGTM with one minor comment.

File: api/v1alpha1/hazelcast_types.go [hunk 2]
```diff
@@ -81,6 +81,11 @@ type HazelcastPersistenceConfiguration struct {
 	// +optional
 	ClusterDataRecoveryPolicy DataRecoveryPolicyType `json:"clusterDataRecoveryPolicy"`
 
+	// AutoForceStart enables the detection of constantly failing cluster and trigger the Force Start action.
+	// +kubebuilder:default:=false
+	// +optional
+	AutoForceStart bool `json:"autoForceStart"`
```
Comment (hakanmemisoglu) [inline api/v1alpha1/hazelcast_types.go:87]: Is `ForceStart DataRecoveryPolicyType` required after this field is introduced?

**Review by hakanmemisoglu — APPROVED** [review]

File: api/v1alpha1/hazelcast_types.go (hunk 2 above)
Comment (hakanmemisoglu) [reply api/v1alpha1/hazelcast_types.go:87]: @SeriyBg @hasancelik

File: api/v1alpha1/hazelcast_types.go [hunk 3]
```diff
@@ -200,6 +207,7 @@ const (
 	MemberAccessLoadBalancer MemberAccess = "LoadBalancer"
 )
 
+// Returns true if ClusterDataRecoveryPolicy is FullRecoveryOnly
```
Comment (hakanmemisoglu) [inline api/v1alpha1/hazelcast_types.go:210]:
```suggestion
// Returns true if ClusterDataRecoveryPolicy is not FullRecoveryOnly
```

**Review by hasancelik — APPROVED** [review]

**Pull request merged**